lib/determine-basal/determine-basal.js

```diff
@@ -374,10 +374,20 @@ var determine_basal = function determine_basal(glucose_status, currenttemp, iob_
             }
         }
     }
+  
+    var minutes_running;
+    if (typeof currenttemp.duration == 'undefined' || currenttemp.duration == 0) {
+        minutes_running = 0;
```

Comment:
This will cause incorrect behavior when duration hits 0: it will cause minutes_running to drop from 29 to 0, and cause lowtempimpact to be calculated incorrectly.  This should default to 30 instead I believe.